workspaces/local-cli/src/shared/coverage-2.ts

```diff
@@ -0,0 +1,126 @@
+//TODO: Figure out where this logic should go
+
+import { IPathMapping } from "@useoptic/cli-config";
+import { getSpecEventsFrom } from "@useoptic/cli-config/build/helpers/read-specification-json";
+import { CaptureInteractionIterator } from "@useoptic/cli-shared/build/captures/avro/file-system/interaction-iterator";
+import { IHttpInteraction } from "@useoptic/domain-types";
+import * as DiffEngine from '@useoptic/diff-engine-wasm/engine/build';
+
+// This concept came from the Scala codebase's "Concern" concept
+// https://github.com/opticdev/optic-core/blob/master/core/optic/shared/src/main/scala/com/useoptic/coverage/CoverageConcerns.scala
+// JS/TS doesn't have the kind of value-types or case-classes that scala does,
+// but we can emulate them like this. I think...
+
+// o.O
+const createDataClass = <P extends Array<any>>(func: (...args: P)=>void): (...args: P)=>string => {
+  return (...args: P) => `${func.name}-${args.map(arg=>JSON.stringify(arg)).join("-")}`
+}
+
+const TotalInteractions = createDataClass(function TotalInteractions() {})
+const TotalUnmatchedPath = createDataClass(function TotalUnmatchedPath() {})
+const TotalForPath = createDataClass(function TotalForPath(pathId: String) {})
+const TotalForPathAndMethod = createDataClass(function TotalForPathAndMethod(pathId: String, httpMethod: String) {})
+const TotalForPathAndMethodAndStatusCode = createDataClass(function TotalForPathAndMethodAndStatusCode(pathId: String, httpMethod: String, httpStatusCode: number) {})
+const TotalForPathAndMethodWithoutBody = createDataClass(function TotalForPathAndMethodWithoutBody(pathId: String, httpMethod: String) {})
+const TotalForPathAndMethodAndContentType = createDataClass(function TotalForPathAndMethodAndContentType(pathId: String, httpMethod: String, requestContentType: String) {})
+const TotalForPathAndMethodAndStatusCodeWithoutBody = createDataClass(function TotalForPathAndMethodAndStatusCodeWithoutBody(pathId: String, httpMethod: String, httpStatusCode: number) {})
+const TotalForPathAndMethodAndStatusCodeAndContentType = createDataClass(function TotalForPathAndMethodAndStatusCodeAndContentType(pathId: String, httpMethod: String, httpStatusCode: number, responseContentType: String) {})
+const TotalForRequest = createDataClass(function TotalForRequest(requestId: string) {})
+const TotalForResponse = createDataClass(function TotalForResponse(responseId: string) {})
+
+type ICoverageMap = {[key: string]: number};
+
+const recordEvent = (evt: string, is_diff: boolean, coverage_map: ICoverageMap) => {
+  const key = `${evt}-${is_diff?'diff':'nodiff'}`;
```

Comment:
i think you should initialize 2 coverage maps, and have the caller provide either the diff one or the non-diff one, as opposed to adding another layer of keys

Reply:
I like that

Reply:
Done